project/ValidatePullRequest.scala

```diff
@@ -29,580 +30,89 @@ import scala.collection.immutable
-        }.flatten ++ filters).forall { f =>
-          if (f(problem)) {
-            true
-          } else {
-            //log(projectName + ": filtered out: " + problem.description + "\n  filtered by: " + f)
-            false
-          }
-        }
-
-        val backErrors = backward filter isReported(module, backwardFilters)
-        val forwErrors = forward filter isReported(module, forwardFilters)
-
-        val filteredCount = backward.size + forward.size - backErrors.size - forwErrors.size
-        val filteredNote = if (filteredCount > 0) " (filtered " + filteredCount + ")" else ""
-
-        // TODO - Line wrapping an other magikz
-        def prettyPrint(p: core.Problem, affected: String): String = {
-          " * " + p.description(affected) + p.howToFilter.map("\n   filter with: " + _).getOrElse("")
-        }
-
-        log(s"$projectName: found ${backErrors.size+forwErrors.size} potential binary incompatibilities while checking against $module $filteredNote")
-        ((backErrors map {p: core.Problem => prettyPrint(p, "current")}) ++
-          (forwErrors map {p: core.Problem => prettyPrint(p, "other")})) foreach { p =>
-          log(p)
-        }
-        backErrors.nonEmpty || forwErrors.nonEmpty
-      }
-
-      def withLogger[T](f: (String => Unit) => T): (String, T) = {
-        val stringWriter = new StringWriter()
-        val printWriter = new PrintWriter(stringWriter)
-
-        val result = f(printWriter.println)
-
-        printWriter.close()
-        stringWriter.close()
-        (stringWriter.toString, result)
-      }
-
-
-      val allResults =
-        mimaPreviousClassfiles.value.toSeq.map {
-          case (moduleId, file) =>
-            val problems = SbtMima.runMima(
-              file,
-              mimaCurrentClassfiles.value,
-              (mimaFindBinaryIssues / fullClasspath).value,
-              mimaCheckDirection.value,
-              scalaVersion.value,
-              streams.value.log,
-              Nil
-            )
-
-            val binary = mimaBinaryIssueFilters.value
-            val backward = mimaBackwardIssueFilters.value
-            val forward = mimaForwardIssueFilters.value
+  override def requires = AkkaHttpValidatePullRequest && MimaPlugin
+  override lazy val projectSettings =
+    CliOptions.mimaEnabled.ifTrue(additionalTasks += mimaReportBinaryIssues).toList
+}
 
-            withLogger { logger =>
-              reportModuleErrors(
-                moduleId,
-                problems._1, problems._2,
-                binary,
-                backward,
-                forward,
-                logger,
-                name.value)
-            }
-        }
+/**
+ * This autoplugin adds Paradox doc generation to validatePullRequest task,
+ * when a project has ParadoxPlugin autoplugin enabled.
+ */
+object ParadoxWithPrValidation extends AutoPlugin {
+  import AkkaHttpValidatePullRequest._
```

Comment:
It might be better to even here be explicit and use `AkkaHttpValidatePullRequest.additionalTasks`